### PR TITLE
Enhance demo loop application

### PR DESCRIPTION
**Pull request by sethrj**
- Output some JSON results even if the run fails
- Fix build on clang due to missing virtual destructor
- Add timers around I/O functions (and small helper class `ScopedTimeLog`)
- Always renormalize rotated angle vectors rather than asserting unit length (@paulromano )
- Add a bunch of small config options for wildstyle demo analysis; these can be removed later
- Change initializer storage from capacity factor to just capacity, since it currently depends on number of primaries as well as number of simultaneous tracks

### CODE REVIEW

File: src/sim/TrackInitData.hh [hunk 1]
```diff
@@ -57,22 +57,22 @@ struct TrackInitParamsData<W, MemSpace::host>
     //// DATA ////
 
     Items<Primary> primaries;     //!< Primary particles
-    size_type storage_factor = 3; //!< Initializer/parent storage per track
+    size_type      capacity = 3;  //!< Initializer/parent storage per track
```
Comment (amandalund) [inline src/sim/TrackInitData.hh:60]: Maybe just remove the default value here.

File: src/sim/TrackInitData.hh [hunk 2]
```diff
@@ -166,7 +166,7 @@ struct ResizableData
  * parents, and \c vacancies are all resizable, and \c track_counters has size
  * \c num_events.
  * - \c initializers stores the data for primaries and secondaries waiting to
- *   be turned into new tracks and can be any size up to \c storage_factor * \c
+ *   be turned into new tracks and can be any size up to \c capacity * \c
  *   num_tracks.
```
Comment (amandalund) [inline src/sim/TrackInitData.hh:170]:
```suggestion
 *   be turned into new tracks and can be any size up to \c capacity.
```

File: src/sim/TrackInitData.hh (hunk 1 above)
Comment (sethrj) [reply src/sim/TrackInitData.hh:60]: Hah, yes. Three is not a very good choice 🙄

File: CMakePresets.json [hunk 3]
```diff
@@ -6,7 +6,7 @@
       "name": "default",
       "displayName": "Automatic options (debug with tests)",
       "description": "Dependencies are enabled based on environment probing",
-      "binaryDir": "${sourceDir}/build",
+      "binaryDir": "${sourceDir}/build-${presetName}",
```
Comment (amandalund) [inline CMakePresets.json:9]: think you might also need to update the build dir in run-ci.sh

Comment (sethrj) [reply CMakePresets.json:9]: Ah hm I need to manually add that to the ci presets then. Thanks.

**Review by amandalund — APPROVED** [review]
Thanks @sethrj!

**Pull request merged**